daffodil-sapi/src/main/scala/org/apache/daffodil/sapi/Daffodil.scala

```diff
@@ -1089,3 +1090,18 @@ class DaffodilUnparseContentHandler private[sapi] (sContentHandler: SDaffodilUnp
   override def skippedEntity(name: String): Unit =
     contentHandler.skippedEntity(name)
 }
+
+/**
+ * Exposes access to the DFDLCatalogResolver to the Java API.
+ *
+ * Resolves URI/URL/URNs to loadable files/streams.
+ *
+ * The DFDLCatalogResolver implements the following interfaces:
+ *  - org.apache.xerces.xni.parser.XMLEntityResolver
+ *  - org.w3c.dom.ls.LSResourceResolver
+ *  - org.apache.sax.EntityResolver
+ *  - org.apache.sax.ext.EntityResolver2
```

Comment:
I'm not sure listing the interfaces is particularly helpful, especially since most people probably don't care too much about them aside from EntityResolver? 

Maybe instead we just change the get function to look something like this:
```scala
def get: EntityResolver = DFDLCatalogResolver.get
```
That way it is self documenting that this returns an `EntityResolver`.

Though, would a user ever need one of the other resolver interfaces? I would guess no? But if so, maybe we have additional functions that return each one that a user might need? E.g.

```
def getEntityResolver: EntityResolver = DFDLCatalogResolver.get
def getXMLEntityResolver: XMLEntityResolver = DFDLCatalogResolver.get
...
```

Seems a bit verbose, and I'm not sure if that makes sense or not?

Maybe another option is just have a class, e.g.

```scala
class DaffodilXMLEntityResolver extends DFDLCatalogResolver
```
And then maybe the javadoc just shows the interfaces that are implemented? I'm not sure if it does or not?

Reply:
I think I like the idea of the multipl getXXXResovler functions.  I believe I needed to use the XMLEntityResolver with one of the EXI implementations, so I definitely needed more than just the SAXEntityResolver interface.